### PR TITLE
chore: refactor exports

### PR DESCRIPTION
**Pull request by hirasso**
**Description**

- Move global declarations to index.ts
- Ensure all type definitions make it into `/dist`
- Fix return type of `swup.getFragmentVisit()` (hopefully)

**Checks**

- [x] The PR is submitted to the `master` branch
- [x] The code was linted before pushing (`npm run lint`)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
 import FragmentPlugin from './SwupFragmentPlugin.js';
 export default FragmentPlugin;
+import type { getFragmentVisit } from './inc/functions.js';
 import type { Route, FragmentVisit } from './inc/defs.js';
 export type { Options, Rule, FragmentElement, FragmentVisit } from './inc/defs.js';
 
 declare module 'swup' {
 	export interface Swup {
-		getFragmentVisit?: (route: Route) => FragmentVisit | undefined;
+		getFragmentVisit?: (route: Route) => ReturnType<typeof getFragmentVisit>;
```
Comment (daun) [inline src/index.ts:9]: Could you just do the whole line with `typeof`?

```ts
getFragmentVisit?: typeof getFragmentVisit;
```

Comment (hirasso) [reply src/index.ts:9]: No, I don't want to expose the logger to users

Comment (hirasso) [reply src/index.ts:9]: Wait, but maybe I don't need to. Let me look into it.

Comment (daun) [reply src/index.ts:9]: I see, in that case that's the only way I guess?

**Review by daun — APPROVED** [review]

**Comment by hirasso** [comment]
I'll just merge an release. It's the only way we can know for sure.

**Pull request merged**